Source/loadsave.cpp

```diff
@@ -207,42 +207,825 @@ BOOL OLoad()
+	CopyChar(tbuff, &pPlayer->_pFireResist);
+	CopyChar(tbuff, &pPlayer->_pLghtResist);
+	CopyInt(tbuff, &pPlayer->_pGold);
+	CopyInt(tbuff, &pPlayer->_pInfraFlag);
+	CopyInt(tbuff, &pPlayer->_pVar1);
+	CopyInt(tbuff, &pPlayer->_pVar2);
+	CopyInt(tbuff, &pPlayer->_pVar3);
+	CopyInt(tbuff, &pPlayer->_pVar4);
+	CopyInt(tbuff, &pPlayer->_pVar5);
+	CopyInt(tbuff, &pPlayer->_pVar6);
+	CopyInt(tbuff, &pPlayer->_pVar7);
+	CopyInt(tbuff, &pPlayer->_pVar8);
+	CopyBytes(tbuff, NUMLEVELS, &pPlayer->_pLvlVisited);
+	CopyBytes(tbuff, NUMLEVELS, &pPlayer->_pSLvlVisited);
+	tbuff += 2;
+	CopyInt(tbuff, &pPlayer->_pGFXLoad);
+
+	tbuff += sizeof(DWORD) * 8;
+	CopyInt(tbuff, &pPlayer->_pNFrames);
+	CopyInt(tbuff, &pPlayer->_pNWidth);
+
+	tbuff += sizeof(DWORD) * 8;
+
+	CopyInt(tbuff, &pPlayer->_pWFrames);
+	CopyInt(tbuff, &pPlayer->_pWWidth);
+
+	tbuff += sizeof(DWORD) * 8;
+
+	CopyInt(tbuff, &pPlayer->_pAFrames);
+	CopyInt(tbuff, &pPlayer->_pAWidth);
+	CopyInt(tbuff, &pPlayer->_pAFNum);
+
+	tbuff += sizeof(DWORD) * 24;
+
+	CopyInt(tbuff, &pPlayer->_pSFrames);
+	CopyInt(tbuff, &pPlayer->_pSWidth);
+	CopyInt(tbuff, &pPlayer->_pSFNum);
+
+	tbuff += sizeof(DWORD) * 8;
+
+	CopyInt(tbuff, &pPlayer->_pHFrames);
+	CopyInt(tbuff, &pPlayer->_pHWidth);
+
+	tbuff += sizeof(DWORD) * 8;
+
+	CopyInt(tbuff, &pPlayer->_pDFrames);
+	CopyInt(tbuff, &pPlayer->_pDWidth);
+
+	tbuff += sizeof(DWORD) * 8;
+
+	CopyInt(tbuff, &pPlayer->_pBFrames);
+	CopyInt(tbuff, &pPlayer->_pBWidth);
+	CopyItems(NUM_INVLOC, pPlayer->InvBody);
+	CopyItems(NUM_INV_GRID_ELEM, pPlayer->InvList);
+	CopyInt(tbuff, &pPlayer->_pNumInv);
+	CopyBytes(tbuff, NUM_INV_GRID_ELEM, pPlayer->InvGrid);
+	CopyItems(MAXBELTITEMS, pPlayer->SpdList);
+	CopyItem(&pPlayer->HoldItem);
+	CopyInt(tbuff, &pPlayer->_pIMinDam);
+	CopyInt(tbuff, &pPlayer->_pIMaxDam);
+	CopyInt(tbuff, &pPlayer->_pIAC);
+	CopyInt(tbuff, &pPlayer->_pIBonusDam);
+	CopyInt(tbuff, &pPlayer->_pIBonusToHit);
+	CopyInt(tbuff, &pPlayer->_pIBonusAC);
+	CopyInt(tbuff, &pPlayer->_pIBonusDamMod);
+	tbuff += 4;
+	CopyInt64(tbuff, &pPlayer->_pISpells);
+	CopyInt(tbuff, &pPlayer->_pIFlags);
+	CopyInt(tbuff, &pPlayer->_pIGetHit);
+
+	CopyChar(tbuff, &pPlayer->_pISplLvlAdd);
+	CopyChar(tbuff, &pPlayer->_pISplCost);
+	tbuff += 2;
+
+	CopyInt(tbuff, &pPlayer->_pISplDur);
+	CopyInt(tbuff, &pPlayer->_pIEnAc);
+	CopyInt(tbuff, &pPlayer->_pIFMinDam);
+	CopyInt(tbuff, &pPlayer->_pIFMaxDam);
+	CopyInt(tbuff, &pPlayer->_pILMinDam);
+	CopyInt(tbuff, &pPlayer->_pILMaxDam);
+	CopyInt(tbuff, &pPlayer->_pOilType);
+
+	CopyChar(tbuff, &pPlayer->pTownWarps);
+	CopyChar(tbuff, &pPlayer->pDungMsgs);
+	CopyChar(tbuff, &pPlayer->pLvlLoad);
+	CopyChar(tbuff, &pPlayer->pBattleNet);
+
+	CopyChar(tbuff, &pPlayer->pManaShield);
+	CopyBytes(tbuff, 3, &pPlayer->bReserved);
+
+	CopyShorts(tbuff, 8, &pPlayer->wReserved);
+
+	CopyInt(tbuff, &pPlayer->pDiabloKillLevel);
+	CopyInts(tbuff, 7, &pPlayer->dwReserved);
+
+	// Omit 10 pointers
+}
+
+void SavePlayer(int i)

```

Comment:
Could you move SavePlayer and SaveItem down to SaveMonster so that they align with the original code.